full/src/main/java/apoc/cypher/CypherExtended.java

```diff
@@ -246,40 +246,46 @@ private void runDataStatementsInTx(
             BlockingQueue<RowResult> queue,
             Map<String, Object> params,
             boolean addStatistics,
-            long timeout,
             boolean reportError,
             String fileName) {
         while (scanner.hasNext()) {
             String stmt = removeShellControlCommands(scanner.next());
             if (stmt.trim().isEmpty()) continue;
-            boolean schemaOperation;
-            try {
-                schemaOperation = isSchemaOperation(stmt);
-            } catch (Exception e) {
-                collectError(queue, reportError, e, fileName);
-                return;
-            }
 
-            if (!schemaOperation) {
-                if (isPeriodicOperation(stmt)) {
-                    Util.inThread(pools, () -> {
-                        try {
-                            return db.executeTransactionally(
-                                    stmt, params, result -> consumeResult(result, queue, addStatistics, tx, fileName));
-                        } catch (Exception e) {
-                            collectError(queue, reportError, e, fileName);
-                            return null;
-                        }
-                    });
-                } else {
+            // Periodic operations cannot be schema operations, so no need to check that here (will fail as invalid
+            // query)
+            if (isPeriodicOperation(stmt)) {
```

Comment:
This check is not fool proof (it's another query string regex). Could we remove this branch?

Reply:
This is 4.4 extended, so I was just leaving the behaviour as it was to not break anything